.github/workflows/generate-parser.yml

```diff
@@ -0,0 +1,41 @@
+# generates the parser with 'tree-sitter generate' if the parser is out of date
+name: Generate Parser
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  generate:
+    name: Generate Parser
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+
+      - name: Cache npm dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install npm dependencies
+        run: npm ci
+
+      - name: Generate parser files
+        run: npx tree-sitter generate
+
+      - name: Commit generated parser files
+        uses: stefanzweifel/git-auto-commit-action@v4
```

Comment:
I'm sure Stefan is great, but I always try to avoid pulling in third-party dependencies if I can. Per [this thread](https://github.community/t/how-does-one-commit-from-an-action/16127/16), I think we could user something like this instead:
```yaml
run: |
  git config --local user.email "$(git log --format='%ae' HEAD^!)"
  git config --local user.name "$(git log --format='%an' HEAD^!)"
  git commit -m "Generate parser"
  git push
```

Reply:
I thought this would be uglier to do with just `git`, but it's very tidy!

Totally understand the third-party-actions averseness, I might steal this for some other grammars now that it's written :smile: